### PR TITLE
Topic6 Aside for unit testing

### LINKED ISSUE #41: Add unit tests to `ArrayStack`
After PR #40 is merged, make an aside for Topic 6 on unit tests. Keep them simple since this is the first time they've seen them. 

I figure going forward, it will be part of each topic, but I'll make the first one an aside that they are to investigate on their own.

### PR DESCRIPTION
**Pull request by jameshughes89**
Brief aside for giving them an intro to unit testing with JUnit in IntelliJ. This closes #41. 

Given that this is intro, it's kept brief and simple, but if there are important gaps, do note them plz.

### CODE REVIEW

**Comment by twentylemon** [comment]
Instead of right click -> open module settings -> mark as tests, you can right click -> mark directory as -> tests. save a modal entirely.

File: src/site/topic6-testing.rst [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+   :align: center
+
+3. Select the ``test`` folder ans select the *Tests*, the green folder, near the top
+
+.. image:: img/test_tests.png
+   :width: 500 px
+   :align: center
+
+4. Create a new class for testing called something appropriate
+    * Perhaps ``TestArrayStack`` for the below example
```
Comment (twentylemon) [inline src/site/topic6-testing.rst:60]: Suffix naming (ie, `ArrayStackTest`) is substantially more common, and also closer to how you'd say it in English. "Array stack tests" is valid, but you'd have to say "tests for array stack" as a prefix.

File: src/site/topic6-testing.rst [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+5. We need to add JUnit to our class path and an easy way to do it is write ``@Test`` in the new class and hit Alt-Enter
+6. Select *Add 'JUnit5....*
+
+.. image:: img/test_altenter.png
+   :width: 500 px
+   :align: center
+
+
+Testing ArrayStack
+==================
```
Comment (twentylemon) [inline src/site/topic6-testing.rst:71]: You may also want to tell them to inject failures just to see how it works. Maybe encourage test writing for their assignments.

File: src/site/topic6-testing.rst [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+=====
+
+* JUnit is a very popular unit testing framework for Java
+* It's easy to use and automates testing for us
+* We will focus on three special things (there is a lot more too it then what will be covered here)
+    * ``@Test`` annotation
+    * ``@DisplayName`` annotation
+    * asserts
+
+* Assuming you are using IntelliJ, here is an easy way to setup JUnit
```
Comment (twentylemon) [inline src/site/topic6-testing.rst:38]: remove _easy_ -- again, it's relative.

File: src/site/topic6-testing.rst (hunk 2 above)
Comment (jameshughes89) [reply src/site/topic6-testing.rst:71]: I think I will save this for Topic8 when I write tests for ``LinkedStack``.

**Review by twentylemon — APPROVED** [review]

**Pull request merged**